File: src/Runtime/Value/ValueString.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {
	public class ValueString : TypeValue<ValueString> {
		private string _value = "";
		public string value { get => _value; }
		public ValueString(string value) {
			_value = value;
		}
		public override string ToString() {
			return _value;
		}
		public override Value Clone() {
			return new ValueString(_value);
		}

	}
} // namespace peak.interpreter

File: src/runtime/value/value_string.cpp
#include "value_string.h"

using namespace peak;

ValueString::ValueString(const std::string& value)
	: _value(value) {
}
const std::string& ValueString::GetValue() const {
	return _value;
}
std::string ValueString::ToString() const {
	return _value;
}
std::string ValueString::ToRawString() const {
	std::string ret;
	ret.reserve(_value.size() + 2);
	ret.append("\"");
	ret.append(_value);
	ret.append("\"");
	return ret;
}
std::shared_ptr<Value> ValueString::Clone() const {
	return std::make_shared<ValueString>(_value);
}
